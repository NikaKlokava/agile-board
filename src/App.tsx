import { Route, Routes } from "react-router";
import "./index.css";
import { BoardPage } from "./pages/board";
import { ErrorPage } from "./pages/error";

function App() {
  return (
    <Routes>
      <Route path="/agile-board" element={<BoardPage />} />
      <Route path="*" element={<ErrorPage />} />
    </Routes>
  );
}

export default App;
