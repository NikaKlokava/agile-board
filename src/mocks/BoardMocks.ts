type MockSubtaskType = {
  name: string;
};

type MockTaskType = {
  name: string;
  description: string;
  subtasks: Array<MockSubtaskType>;
};

export const MockTask: MockTaskType = {
  name: "Create something",
  description: "... it's very important ...",
  subtasks: [
    { name: "while doing 1..." },
    { name: "while doing 2..." },
    { name: "while doing 3..." },
  ],
};

export const MockBoards = [
  {
    name: "Platform Launch",
    board_column: [
      { title: "Todo", tasks: Array.from({ length: 4 }, () => MockTask) },
      { title: "Doing", tasks: Array.from({ length: 6 }, () => MockTask) },
      { title: "Done", tasks: Array.from({ length: 2 }, () => MockTask) },
    ],
  },
  {
    name: "Marketing Plan",
    board_column: [
      { title: "Todo", tasks: Array.from({ length: 7 }, () => MockTask) },
      { title: "Doing", tasks: Array.from({ length: 2 }, () => MockTask) },
      { title: "Done", tasks: Array.from({ length: 4 }, () => MockTask) },
    ],
  },
  {
    name: "Roadmap",
    board_column: [
      { title: "Now", tasks: Array.from({ length: 4 }, () => MockTask) },
      { title: "Next", tasks: Array.from({ length: 6 }, () => MockTask) },
    ],
  },
];
