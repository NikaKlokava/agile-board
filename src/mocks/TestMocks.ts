export const MockTestAddBoard = {
  name: "test name",
  columns: [{ title: "test title", uuid: "test uuid" }],
  usersEmail: ["test"],
  time: 0,
  uuid: "test",
};
