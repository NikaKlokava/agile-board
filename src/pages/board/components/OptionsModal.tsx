import classes from "classnames";
import cl from "./styles/options_modal.module.css";

export const OptionsModal = () => {
  return (
    <div className={cl.options_modal_wrapper}>
      <p className={classes(cl.option_text, cl.edit)}>Edit Boards</p>
      <p className={classes(cl.option_text, cl.delete)}>Delete Boards</p>
    </div>
  );
};
