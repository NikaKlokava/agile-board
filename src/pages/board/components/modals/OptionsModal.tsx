import classes from "classnames";
import cl from "./modal_styles.module.css";

type Props = {
  type: "Board" | "Task";
  onEditClick?: () => void;
  onDeleteClick?: () => void;
};

export const OptionsModal = ({ type, onEditClick, onDeleteClick }: Props) => {
  return (
    <div className={cl.options_modal_wrapper} data-testid="options-modal">
      <p
        className={classes(cl.option_text, cl.edit)}
        data-testid="options-edit-board"
        onClick={onEditClick}
      >
        {`Edit ${type}`}
      </p>
      <p
        className={classes(cl.option_text, cl.delete)}
        data-testid="options-delete-board"
        onClick={onDeleteClick}
      >
        {`Delete ${type}`}
      </p>
    </div>
  );
};
