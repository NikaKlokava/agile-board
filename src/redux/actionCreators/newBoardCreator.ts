export const addBoard = ({ name, columns }: AddBoardPayloadType) => {
  return {
    type: "ADD_BOARD_ACTION",
    payload: {
      name,
      columns,
    },
  };
};

export const selectBoard = (board: BoardType) => {
  return {
    type: "SELECT_BOARD_ACTION",
    payload: {
      board,
    },
  };
};

export const addNewTask = ({
  boardUuid,
  columnUuid,
  title,
  description,
  subtasks,
}: AddTaskPayloadType) => {
  return {
    type: "ADD_NEW_TASK_ACTION",
    payload: {
      boardUuid,
      columnUuid,
      title,
      description,
      subtasks,
    },
  };
};

export const checkSubtask = (subtaskUuid: string) => {
  return {
    type: "CHECK_SUBTASK_ACTION",
    payload: {
      subtaskUuid,
    },
  };
};

export const addNewColumn = (
  uuid: string,
  name: string,
  columns: ColumnsType
) => {
  return {
    type: "ADD_NEW_COLUMN_ACTION",
    payload: {
      uuid,
      name,
      columns,
    },
  };
};

export const deleteBoard = (uuid: string) => {
  return {
    type: "DELETE_BOARD_ACTION",
    payload: {
      uuid,
    },
  };
};

export const deleteACtiveBoard = () => {
  return {
    type: "DELETE_ACTIVE_BOARD_ACTION",
  };
};

export const moveTask = (taskUuid: string, columnUuid: string) => {
  return {
    type: "MOVE_TASK_ACTION",
    payload: { taskUuid, columnUuid },
  };
};
