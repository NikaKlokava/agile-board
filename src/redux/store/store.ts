import { legacy_createStore } from "redux";
import { boardsReducer } from "../reducers/boardReducer";

const store = legacy_createStore(boardsReducer);

export default store;

// export type RootState = ReturnType<typeof store.getState>

// export interface RootState {
//     user: UserState;
//     loader: LoaderState;
//   }
// export type AppDispatch = typeof store.dispatch
