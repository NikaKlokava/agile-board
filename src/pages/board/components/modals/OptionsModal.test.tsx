import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import App from "../../../../App";
import { MockTestAddBoard } from "../../../../mocks/TestMocks";
import { addBoard } from "../../../../redux/actionCreators/newBoardCreator";
import store from "../../../../redux/store/store";
import { OptionsModal } from "./OptionsModal";

afterEach(() => {
  jest.clearAllMocks();
});

describe("Test the OptionsModal component", () => {
  test("The OptionsModal renders correctly", () => {
    jest.mock("./OptionsModal.tsx");
    const optionsModalSnap = renderer
      .create(
        <Provider store={store}>
          <OptionsModal />
        </Provider>
      )
      .toJSON();
    expect(optionsModalSnap).toMatchSnapshot();
  });

  test("The EditBoardModal should be visible by click on edit board", () => {
    render(
      <MemoryRouter initialEntries={["/agile-board"]}>
        <App />
      </MemoryRouter>
    );

    act(() => {
      store.dispatch(addBoard(MockTestAddBoard));
    });

    const optionsIcon = screen.getByTestId("options-icon");

    act(() => {
      optionsIcon.click();
    });

    const editBoardEl = screen.getByTestId("options-edit-board");

    act(() => {
      editBoardEl.click();
      store.dispatch(addBoard(MockTestAddBoard));
    });

    const editBoardModal = screen.getByTestId("edit-board-modal");

    expect(editBoardModal).toBeInTheDocument();
  });

  test("The DeleteBoardModal should be visible by click on delete board", () => {
    render(
      <MemoryRouter initialEntries={["/agile-board"]}>
        <App />
      </MemoryRouter>
    );
    act(() => {
      store.dispatch(addBoard(MockTestAddBoard));
    });
    
    const optionsIcon = screen.getByTestId("options-icon");

    act(() => {
      optionsIcon.click();
    });

    const deleteBoardEl = screen.getByTestId("options-delete-board");

    act(() => {
      deleteBoardEl.click();
    });

    const deleteBoardModal = screen.queryByTestId("delete-board");
    expect(deleteBoardModal).toBeInTheDocument();
  });
});
