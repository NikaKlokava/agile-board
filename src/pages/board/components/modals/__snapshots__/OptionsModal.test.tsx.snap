// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the OptionsModal component The OptionsModal renders correctly 1`] = `
<div
  className="options_modal_wrapper"
  data-testid="options-modal"
>
  <p
    className="option_text edit"
    data-testid="options-edit-board"
  >
    Edit Board
  </p>
  <p
    className="option_text delete"
    data-testid="options-delete-board"
  >
    Delete Board
  </p>
</div>
`;
